client/app/react-components/PlotlyChart.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import createPlotlyComponent from 'react-plotly.js/factory';
+import Plotly from 'plotly.js';
+import bar from 'plotly.js/lib/bar';
+import pie from 'plotly.js/lib/pie';
+import histogram from 'plotly.js/lib/histogram';
+import box from 'plotly.js/lib/box';
+import { each, isArray, isObject } from 'lodash';
+import { normalizeValue, updateData, prepareData, prepareLayout } from '@/visualizations/chart/plotly/utils';
+
+
+Plotly.register([bar, pie, histogram, box]);
+Plotly.setPlotConfig({
+  modeBarButtonsToRemove: ['sendDataToCloud'],
+});
+
+const Plot = createPlotlyComponent(Plotly);
+
+
+const timeSeriesToPlotlySeries = (ss) => {
+  const x = [];
+  const ys = {};
+  each(ss, (series) => {
+    ys[series.name] = [];
+    each(series.data, (point) => {
+      x.push(normalizeValue(point.x));
+      ys[series.name].push(normalizeValue(point.y));
+    });
+  });
+  return [x, ys];
+};
+
+export default class PlotlyChart extends React.Component {
+  static propTypes = {
+    // XXX make this required after porting next layer up
+    options: PropTypes.object,
+    // eslint-disable-next-line react/no-unused-prop-types
+    series: PropTypes.array.isRequired,
+    customCode: PropTypes.string,
+
+  }
+
+  static defaultProps = { options: null, customCode: null };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+      layout: null,
+      revision: 0,
+      x: null,
+      ys: null,
+    };
+    this.refreshCustom = this.refreshCustom.bind(this);
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (!nextProps.options) { return {}; }
```

Comment:
`getDerivedStateFromProps` should return `null` if state doesn't need to be updated.